lib/zendesk_api/collection.rb

```diff
@@ -253,7 +256,7 @@ def next
       if @options["page"]
         clear_cache
         @options["page"] += 1
-      elsif @query = @next_page
+      elsif (@query = @next_page)
```

Comment:
because `=` (assignment) can be confused with `==`, Ruby has a convention of adding parenthesis to let the reader know this was a conscious choice.

Personally, I prefer to avoid using assignments in conditionals.